core/metric.go

```diff
@@ -64,6 +67,31 @@ func (n Namespace) Strings() []string {
 	return ns
 }
 
+// getSeparator returns an valid separator from nsPriorityList.
+// Otherwise the core separator is returned.
+func (n Namespace) getSeparator() string {
+	mp := n.toMap()
+	for _, r := range nsPriorityList {
+		if ok := mp[r]; !ok {
+			return r
+		}
+	}
+	return Separator
+}
+
+// toMap converts namespace values into a character map.
+func (n Namespace) toMap() map[string]bool {
+	m := map[string]bool{}
+	for _, elem := range n {
+		rs := elem.Value
+		for _, r := range rs {
+			s := fmt.Sprintf("%c", r)
```

Comment:
Maybe change `nsPriorityList` to a slice of rune so you don't deal with that line.